build.sh

```diff
@@ -28,10 +32,17 @@ case "$ARCH" in
 		export GOARM=7
 	;;
 esac
+
 # compile Go
 echo $GOARM
-cd go/src \
-	&& git checkout go$GOLANG_VERSION \
+cd go && git checkout COMMIT_$GOLANG_VERSION
+
+if version_le "1.5.2" $GOLANG_VERSION; then
+	cp /patches/golang-$GOLANG_VERSION.patch /go/golang-$GOLANG_VERSION.patch
```

Comment:
Is this needed? I think you can do `patch -p1 < /patches/golang-$GOLANG_VERSION.patch` directly